history.go

```diff
@@ -52,25 +57,33 @@ func Run() error {
 		if err != nil {
 			return fmt.Errorf("error reading the input")
 		}
-		//fmt.Println(text)
 		if text == "exit" || text == "quit" {
 			break
 		}
+		cmdHistory = append(cmdHistory, fmt.Sprintf("[%d] command: %s", time.Now().Unix(), text))
 		output, err := RunCommand(strings.Split(text, " ")[0], strings.Split(text, " ")[1:])
 		if err != nil {
 			fmt.Printf("[ERROR] cannot run command %s: %v\n", text, err)
+			cmdHistory = append(cmdHistory, fmt.Sprintf("[%d] error: %s", time.Now().Unix(), err.Error()))
+			continue
 		}
 		fmt.Printf(output)
-		cmdHistory = append(cmdHistory, text)
-		// switch text {
-		// case "exit\n":
-		// 	break
-		// case "quit\n":
-		// 	break
-		// default:
-		// 	RunCommand(text, nil)
-		// }
+		cmdHistory = append(cmdHistory, fmt.Sprintf("[%d] output: %s", time.Now().Unix(), output))
 	}
 	WriteFile(".history", cmdHistory)
 	return nil
 }
+
+// HandleTerminationSignal creates a 'listener' on a new goroutine which will notify the
+// program if it receives an interrupt from the OS. We then handle this by calling
+// our clean up procedure and exiting the program.
+func HandleTerminationSignal(cmdHistory *[]string) {
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-c
+		fmt.Println("\r- Ctrl+C pressed in Terminal")
```

Comment:
Well, not necessarily—someone might have killed the process with a signal, for example.